### ISSUE TITLE
Could not reproduce exp in the paper.

### ISSUE THREAD

**Issue by liayan** [iDC-NEU/NeutronBench#2]
hello, I am trying to reproduce your exp in the paper,  hit two issues here wondering if could get help from you.

1. amazon datasets
    python ./generate_nts_dataset.py --dataset amazon
   no adj_full.npz and neats.npz and so on.
   There is an Amazon dataset in the google drive, however no such file either

2. Adaptive Batch Size

  I went through all your exp code files but did not find the command line for it. Could you elaborate more?

Looking forward to your reply.

**Comment by Sanzo00** [comment]
Thank you for your interest in our work! Below are our responses to your questions:

> Issue 1: Missing adj_full.npz and neats.npz in the Amazon dataset

We have provided the processed Amazon dataset (`amazon.zip`) in [Google Drive](https://drive.google.com/drive/folders/1rkvInbkz3vGFiqS64xDc8Nt7uEmhNqIi), including `amazon.mask`, `amazon.label`, `amazon.edge`, and `amazon.feat`. These files are in the required format for NeutronBench and can be used directly.

If you prefer to generate the dataset yourself using `generate_nts_dataset.py`, you need to download the original Amazon dataset first from the following link: [Google Drive - Amazon Original Dataset](https://drive.google.com/drive/folders/1uc76iCxBnd0ntNliosYDHHUc_ouXv9Iv)

> Issue 2: Adaptive Batch Size Experiment

We uploaded the adaptive batch size experiment configuration file, and you can reproduce it using the following steps:

```bash
# move to the batch size experiment directory
cd NeutronBench/exp/exp-batch-size

# Run the configuration files
./run_nts.sh 1 ogbn-products-mix.cfg > ogbn-products-mix.log
./run_nts.sh 1 reddit-mix.cfg > reddit-mix.log
```

Key Parameters in configuration file:
1. `BATCH_SIZE_VEC`: Specifies the range of batch sizes.

2. Switching Strategies (two options available):
   - `BATCH_SWITCH_ACC`: If validation accuracy does not increase in a certain epoch, switch to next batch size.
   - `BATCH_SWITCH_TIME`: Switch batch size at fixed intervals.

We hope this helps! Let us know if you have any further questions.

**Comment by liayan** [comment]
Thanks for the quick response.

The amazon data is for partition exp, I need origninal data for dgl partition, did not find a way to partition processed dataset.

 For distdgl training, have you used GPU(backend gloo or others)? It does not matter since the focus is the partition method, I just want to align with your paper. 

I also saw a exp-partition implemented in nts,  what is the difference with Partition/partition? 
BTW, I saw it tried to read files from partition_result, 

read_file /workspace/NeutronBench/exp/Partition/partition/partition_result/metis1-ogbn-products-part4.txt
but only have metis1-ogbn-products-part4 there. 
How could generate part4.txt?

**Comment by liayan** [comment]
Another issue is that dgldist is always stuck on evaluate_test. Is this expected? It could run dgldist from the upstream version in the same environment. 

            if g.rank() == 0:
                dist.recv(tensor=tensor_tmp, src=1)
                tensor_sum += tensor_tmp

                result_arry = (tensor_sum + tensor_0).numpy()
                print(result_arry)
                all_val_acc = result_arry[0][0] / result_arry[0][1]
                print('All val acc:', all_val_acc)
            else:
                dist.send(tensor=th.tensor([val_right_num.item(),
                                            val_sum_num]),
                          dst=0)

**Comment by Sanzo00** [comment]
Sorry for the late reply — I’ve been occupied with a paper deadline recently.

Regarding your question:

We use DistDGL as the experimental platform for graph partitioning. For details on setting up DistDGL, you can refer to the following documentation:
- [Chapter 7: Distributed Training](https://www.dgl.ai/dgl_docs/guide/distributed.html)
- [Distributed Training Examples](https://github.com/dmlc/dgl/tree/master/examples/distributed/graphsage)

We also provide code for distributed training at: [`exp/Partition/partition/train_dist.py`](https://github.com/iDC-NEU/NeutronBench/blob/main/exp/Partition/partition/train_dist.py), which supports GCN and GraphSAGE in distributed settings.

To use DistDGL for distributed training, you need to first generate the graph partition files. We support multiple partitioning methods in [exp/Partition/partition/partition_graph.py](https://github.com/iDC-NEU/NeutronBench/blob/main/exp/Partition/partition/partition_graph.py), including Hash, Metis-based, and Streaming-based approaches (e.g., PaGraph, ByteGNN). You can generate the partition files using the following commands:

> Generate Graph Partition Files

```bash
# For Hash, Metis-based, ParaGraph, and ByteGNN (Python implementation)
python partition_graph.py --dataset ogbn-arxiv --num_parts 4 --mode hash

# For the ByteGNN partition method (C++ implementation for large graphs)
bash bytegnn_partition.sh
```
